frontend/src/components/v2/CopyButton/CopyButton.tsx

```diff
@@ -0,0 +1,57 @@
+import { faCheck, faCopy, IconDefinition } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { twMerge } from "tailwind-merge";
+
+import { useTimedReset } from "@app/hooks";
+
+import { IconButton } from "../IconButton";
+import { Tooltip } from "../Tooltip";
+
+export type CopyButtonProps = {
+  value: string;
+  size?: "xs" | "sm" | "md" | "lg";
+  variant?: "solid" | "outline" | "plain" | "star" | "outline_bg";
+  color?: string;
+  name?: string;
+  icon?: IconDefinition;
+};
+
+export const CopyButton = ({
+  value,
+  size = "sm",
+  variant = "solid",
+  color,
+  name,
+  icon = faCopy
+}: CopyButtonProps) => {
+  const tooltipText = name ? `Copy ${name}` : "Copy to clipboard";
```

Comment:
Nit: How about we move this constant into the initialState?

```typescript
useTimedReset<string>({
    initialState: name ? `Copy ${name}` : "Copy to clipboard"
  });
```

Reply:
ah yeah!
great idea
cant think why I didnt do that